tests/dvslib/dvs_database.py

```diff
@@ -0,0 +1,397 @@
+        `DEFAULT_POLLING_CONFIG`. Users can specify their own PollingConfig,
+        but this shouldn't typically be necessary.
+    """
+    DEFAULT_POLLING_CONFIG = PollingConfig(polling_interval=0.01, timeout=5, strict=True)
+
+    def __init__(self, db_id, connector):
+        """
+            Initializes a DVSDatabase instance.
+
+            Args:
+                db_id (int): The integer ID used to identify the given database
+                    instance in redis.
+                connector (str): The I/O connection used to communicate with
+                    redis (e.g. unix socket, tcp socket, etc.).
+
+            NOTE: Currently it's most convenient to let the user specify the
+            connector since it's set up in the dvs fixture. We may abstract
+            this further in the future as we refactor dvs.
+        """
+
+        self.db_connection = swsscommon.DBConnector(db_id, connector, 0)
+
+    def create_entry(self, table_name, key, entry):
+        """
+            Adds the mapping {`key` -> `entry`} to the specified table.
+
+            Args:
+                table_name (str): The name of the table to add the entry to.
+                key (str): The key that maps to the entry.
+                entry (Dict[str, str]): A set of key-value pairs to be stored.
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+        formatted_entry = swsscommon.FieldValuePairs(entry.items())
+        table.set(key, formatted_entry)
+
+    def get_entry(self, table_name, key,
+                  polling_config=DEFAULT_POLLING_CONFIG):
+        """
+            Gets the entry stored at `key` in the specified table.
+
+            Args:
+                table_name (str): The name of the table where the entry is
+                    stored.
+                key (str): The key that maps to the entry being retrieved.
+                polling_config (PollingConfig): The parameters to use to poll
+                    the db.
+
+            Returns:
+                Dict[str, str]: The entry stored at `key`. If no entry is found,
+                then an empty Dict will be returned.
+
+        """
+
+        access_function = self._get_entry_access_function(table_name, key, True)
+        return self._db_poll(polling_config, access_function)
+
+    def delete_entry(self, table_name, key):
+        """
+            Removes the entry stored at `key` in the specified table.
+
+            Args:
+                table_name (str): The name of the table where the entry is
+                    being removed.
+                key (str): The key that maps to the entry being removed.
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+        table._del(key)  # pylint: disable=protected-access
+
+    def is_entry_empty(self,
+                       table_name,
+                       key,
+                       polling_config=DEFAULT_POLLING_CONFIG):
```

Comment:
Want to call out this function specifically. It gives a sense of Boolean check but has a polling implemented.

Reply:
That's a good point, I went ahead and updated it. Let me know if the new names make sense!